inspirehep/modules/workflows/tasks/matching.py

```diff
@@ -212,6 +212,34 @@ def _is_core(record):
         obj.extra_data['core'] = True
 
 
+def set_wf_not_completed_ids_to_wf(obj, skip_blocked=True):
+    """Return list of all matched workflows ids which are not completed
+    also stores this list to wf.extra_data['holdingpen_matches']
+    By default it do not show workflows which are already blocked by this workflow
+
+    Arguments:
+        obj: a workflow object.
+        skip_blocked: boolean, if True do not returnd workflows blocked by this one,
+            if False then return all matched workflows.
+    """
+
+    def _non_completed(base_record, match_result):
+        return not get_value(match_result,
+                             '_source._workflow.status') == 'COMPLETED'
+
+    def is_workflow_blocked_by_another_workflow(workflow_id):
+        workflow = workflow_object_class.get(workflow_id)
+        return obj.id in workflow.extra_data.get('holdingpen_matches', [])
+
+    matched_ids = pending_in_holding_pen(obj, _non_completed)
+    if skip_blocked:
+        matched_ids = [_id for _id in matched_ids if
+                       not is_workflow_blocked_by_another_workflow(_id)]
+    obj.extra_data['holdingpen_matches'] = matched_ids
+    obj.save()
```

Comment:
I was thinking that we can use the `save_workflow` function, which also commits to the DB, this one just indexes to ES (and we try also to unify the way we do it).